apps/test/unit/code-studio/components/header/SignInCalloutTest.js

```diff
@@ -1,10 +1,13 @@
 import React from 'react';
 import SignInCalloutWrapper from '@cdo/apps/code-studio/components/header/SignInCalloutWrapper';
-import {shallow} from 'enzyme';
+import {shallow, mount} from 'enzyme';
 import {expect} from '../../../../util/reconfiguredChai';
 import i18n from '@cdo/locale';
+import sinon from 'sinon';
+import cookies from 'js-cookie';
 
 const wrapper = shallow(<SignInCalloutWrapper />);
+const parentWrapper = mount(<SignInCalloutWrapper />);
```

Comment:
Could you say a bit about why we need both a shallow() and mount() versions of this component?  Also, what does "parent" in the name refer to?

Reply:
Yes! mount() allows for a click simulation, but shallow suffices for the other use cases. Parent was mostly to differentiate, and to signify that this wrapper was a deeper/more useful one. That said- it could've been called anything else!

Reply:
I just realized...  in some other tests, the wrapper is instantiated inside each test.  Does instantiating it up here make it more likely that one test will contaminate another?  (Moving it into each test also helps resolve's the naming problem because you can just call it wrapper everywhere.)

Reply:
Good point. I'll just do that!